spring-cloud-azure-config/src/test/java/com/microsoft/azure/spring/cloud/config/ConfigResourceManagerTest.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.azure.spring.cloud.config;
+
+import com.microsoft.azure.Page;
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.credentials.AzureTokenCredentials;
+import com.microsoft.azure.management.Azure;
+import com.microsoft.azure.management.resources.GenericResource;
+import com.microsoft.azure.management.resources.GenericResources;
+import com.microsoft.azure.management.resources.Subscription;
+import com.microsoft.azure.management.resources.Subscriptions;
+import com.microsoft.azure.spring.cloud.config.msi.ConfigMSICredentials;
+import com.microsoft.azure.spring.cloud.config.msi.ConfigResourceManager;
+import com.microsoft.azure.spring.cloud.context.core.util.Tuple;
+import com.microsoft.rest.RestException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PowerMockIgnore;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Arrays;
+
+import static com.microsoft.azure.spring.cloud.config.TestConstants.*;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.powermock.api.mockito.PowerMockito.when;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest(Azure.class)
+@PowerMockIgnore({"javax.net.ssl.*"})
+public class ConfigResourceManagerTest {
+    @Mock
+    private Azure.Authenticated authenticated;
+
+    @Mock
+    private ConfigMSICredentials msiCredentials;
+
+    @Mock
+    private Subscriptions subscriptions;
+
+    @Before
+    public void setup() {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    @Test
+    public void configStoreCanBeSearchedIfExist() {
```

Comment:
Suggest to separate this test in different part if possible, it is a little long.

Reply:
Good point, move some data initialization into another method.